application/forms/Config/General/ApplicationConfigForm.php

```diff
@@ -70,57 +70,25 @@ public function createElements(array $formData)
             )
         );
 
-        // we do not need this form for setup because we set the database there as default.
-        // this form is only displayed in configuration -> application if preferences backend type of ini is recognized
-        if (isset($formData['global_config_backend']) && $formData['global_config_backend'] === 'ini') {
-            $this->addElement(
-                'select',
-                'global_config_backend',
-                [
-                    'required'      => true,
-                    'autosubmit'    => true,
-                    'label'         => $this->translate('User Preference Storage Type'),
-                    'multiOptions'  => [
-                        'ini'   => $this->translate('File System (INI Files)'),
-                        'db'    => $this->translate('Database')
-                    ]
-                ]
-            );
-        } else {
-            $this->addElement(
-                'hidden',
-                'global_config_backend',
-                [
-                    'required'  => true,
-                    'value'     => 'db',
-                    'disabled'  => true
-                ]
-            );
+        $backends = array();
+        foreach (ResourceFactory::getResourceConfigs()->toArray() as $name => $resource) {
+                $backends[$name] = $name;
```

Comment:
Hmmm, just use `array_keys` here and below, where the options are populated, transform it with `array_combine($backends, $backends)`